version: "3.9"
services:
  api:
    build:
      context: .
    image: pulsercoder/skypro_pkuzmin_app
    networks:
      - my-network
    ports:
      - 80:80
    depends_on:
      pg:
        condition: service_healthy
    volumes:
      - ./docker-config.py:/code/config.py
  pg:
    image: postgres:latest
    networks:
      - my-network
    environment:
      POSTGRES_USER: flask_app
      POSTGRES_PASSWORD: flask_app_password
      POSTGRES_DB: flask_app
    volumes:
      - ./pg_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 10s
      retries: 5
networks:
  my-network:
